ext/drreg/drreg.c

```diff
@@ -402,7 +402,11 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
         /* Restore aflags to app value */
         LOG(drcontext, LOG_ALL, 3, "%s @%d."PFX" aflags=0x%x: lazily restoring aflags\n",
             __FUNCTION__, pt->live_idx, instr_get_app_pc(inst), aflags);
-        res = drreg_restore_aflags(drcontext, bb, inst, pt, false/*keep slot*/);
+        /* On AArch64, we cannot keep the slot, because spill_reg is used to save the
+         * aflags and this functions requires the slot to be released.
+         */
+        res = drreg_restore_aflags(drcontext, bb, inst, pt,
+                                   IF_AARCHXX_ELSE(true, false/*keep slot*/));
```

Comment:
I think the problem is the following:
On AArch64, `drreg_spill_aflags` uses `spill_reg` with a scratch register, which can be different in different calls to `drreg_spill_aflags` (https://github.com/DynamoRIO/dynamorio/blob/master/ext/drreg/drreg.c#L1123). When `spill_reg` is called with the same slot but different registers, the following assertion will fail: https://github.com/DynamoRIO/dynamorio/blob/master/ext/drreg/drreg.c#L184 .

By releasing the slot in `drreg_restore_aflags`, the problem can be prevented.

I think this problem surfaced because when `*usage = EFLAGS_WRITE_ALL | EFLAGS_READ_ALL` is used for all AArch64 instructions, the aflags may have to be restored multiple times in a basic block, whereas before they would only be restored at the end (I think).